crates/autopilot/src/decoded_settlement.rs

```diff
@@ -265,47 +251,56 @@ impl DecodedSettlement {
         })
     }
 
-    /// Returns unsubsidized fees for all trades.
-    /// Length of the returned vector is equal to the length of `self.trades`
-    /// and `order_fees`.
+    /// Returns fees for all trades.
     pub fn all_fees(
         &self,
         external_prices: &ExternalPrices,
-        order_fees: &[(OrderUid, Option<U256>)],
+        domain_separator: &DomainSeparator,
     ) -> Vec<Fees> {
         self.trades
             .iter()
-            .zip(order_fees.iter())
-            .map(|(trade, (order, order_fee))| {
-                self.fee(external_prices, *order, *order_fee, trade)
-                    .unwrap_or_else(|| {
-                        tracing::warn!("possible incomplete fee calculation");
-                        // we should have an order execution for every trade
-                        Fees {
-                            order: *order,
-                            sell: U256::zero(),
-                            native: U256::zero(),
-                        }
-                    })
+            .filter_map(|trade| {
+                let order = match trade.uid(domain_separator, &self.tokens) {
+                    Ok(order) => order,
+                    Err(err) => {
+                        tracing::error!(
+                            ?err,
+                            ?trade,
+                            "failed to calculate order uid, we don't know which order this trade \
+                             belongs to"
+                        );
+                        return None;
+                    }
+                };
+
+                Some(self.fee(trade, order, external_prices).unwrap_or_else(|| {
+                    tracing::warn!("possible incomplete fee calculation");
+                    // we should have an order execution for every trade
+                    Fees {
+                        order,
+                        kind: FeeKind::User,
```

Comment:
nit: It's only really possible for the `FeeKind::Surplus` case to fail so I think it would be more appropriate to return that kind here instead.

Reply:
Fixed.